bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java

```diff
@@ -80,6 +80,25 @@
     void startScan(ScanListener listener);
 
     /**
+     * Triggers this service to start an active scan for new devices.<br>
+     * This method must not block any calls such as {@link #abortScan()} and
+     * must return fast.
+     * <p>
+     * If started, any registered {@link DiscoveryListener} must be notified about {@link DiscoveryResult}s.
+     * <p>
+     * If there is already a scan running, it is aborted and a new scan is triggered.
+     * <p>
+     * In contrast to {@link #startScan(ScanListener)} this method can be given a starting time stamp
```

Comment:
I agree with you that it's not a good idea, to have it on the public interface. But I need a way to call the startScan method from the DiscoveryServiceRegistry with the given timestamp. I see no other way, how the registry could handle this itself. Can we assume, that each DiscoveryService is at least an AbstractDiscoveryService? Then this method doesn't need to be on the interface, only on the abstract class and the registry could use it. But I think, we cannot, as the user might be free to implement the interface completely by itself. 

The other question, whether it's legal or not, that a discovery service removes results that might not have been created by itself, is a good one. However, this would require to mark each result from which service it has been created (or updated lately). Even when implementing it this way, how could we be sure, that no one else has overridden the removeOlderResult method in another DiscoveryService to handle things differently? We should have made this method final...